elasticdl/python/data/data_reader.py

```diff
@@ -165,6 +165,59 @@ def _get_reader(self, table_name):
     def _get_odps_table_name(shard_name):
         return shard_name.split(":")[0]
 
+    def default_dataset_fn(self):
+        _check_required_kwargs(["label_col"], self._kwargs)
+
+        def dataset_fn(dataset, mode, metadata):
+            def _parse_data(record):
+                label_col_name = self._kwargs["label_col"]
+                record = tf.strings.to_number(record, tf.float32)
+
+                def _get_features_without_labels(
+                    record, label_col_ind, features_shape
+                ):
+                    features = [
+                        record[:label_col_ind],
+                        record[label_col_ind + 1 :],  # noqa: E203
+                    ]
+                    features = tf.concat(features, -1)
+                    return tf.reshape(features, features_shape)
+
+                features_shape = (len(metadata.column_names) - 1, 1)
+                labels_shape = (1,)
+                if mode != Mode.PREDICTION:
+                    if label_col_name not in metadata.column_names:
+                        raise ValueError(
+                            "Missing the label column '%s' in the retrieved "
```

Comment:
It would be better to add the mode in the Exception message

Reply:
Added